pkg/settlement/swap/chequebook/init.go

```diff
@@ -48,9 +49,13 @@ func checkBalance(
 			return err
 		}
 
-		gasPrice, err := swapBackend.SuggestGasPrice(timeoutCtx)
-		if err != nil {
-			return err
+		gasPrice := sctx.GetGasPrice(ctx)
+
+		if gasPrice == nil {
+			gasPrice, err = swapBackend.SuggestGasPrice(timeoutCtx)
+			if err != nil {
+				return err
+			}
 		}
 
 		minimumEth := gasPrice.Mul(gasPrice, big.NewInt(250000))
```

Comment:
Why are you multiplying with 250k here? Is this needed?

Reply:
because we want to be able to use around 250k gas. For the deployment and the initial top up transaction.